### PR TITLE
Move AsciiDocs from javadoc to @Documented value

### PR DESCRIPTION
**Pull request by lutovich**

### CODE REVIEW

**Comment by lutovich** [comment]
Small AsciiDocs look quite nice inside `@Documented` annotation but big chunks like this https://github.com/lutovich/neo4j/commit/a2909e5ade3cbe4921e4d0b41ec093622ad3a988#diff-ee5a9c98d32445d9a1fbd73e4c12b7c7R51 look extremely ugly. I'm not entirely sure they are more ugly than corresponding javadocs but they are ugly :)

This change makes it possible to build docs on IBM JDK. This is the main motivation for the PR.
Maybe it could be the first step towards moving big chunks of AsciiDoc in separate files and having `@Documented` reference them?

@nawroth what do you think about this PR?

File: community/server/src/test/java/org/neo4j/server/rest/CreateRelationshipDocTest.java [hunk 1]
```diff
@@ -34,18 +34,13 @@
...
 {
-    /**
-     * Upon successful creation of a relationship, the new relationship is
-     * returned.
-     */
     @Test
     @Graph( "Joe knows Sara" )
-    @Documented
+    @Documented( "Upon successful creation of a relationship, the new relationship is returned." )
     @Title( "Create a relationship with properties" )
```
Comment (MishaDemianenko) [inline community/server/src/test/java/org/neo4j/server/rest/CreateRelationshipDocTest.java:44]: @lutovich @nawroth content in documentation already can contain markup, why do we need to use another annotation for `Title` in some of the cases?

Comment (lutovich) [inline community/server/src/test/java/org/neo4j/server/rest/CreateRelationshipDocTest.java:44]: @MishaDemianenko `@Title` indeed feels unnecessary. It's value is used in some dark corners of the docs framework. We should consider removing it in 3.0 I think.

**Comment by lutovich** [comment]
@nawroth I think all docs are fixed now.

File: community/server/src/test/java/org/neo4j/server/rest/CreateRelationshipDocTest.java (hunk 1 above)
Comment (nawroth) [inline community/server/src/test/java/org/neo4j/server/rest/CreateRelationshipDocTest.java:44]: I think we should do as little as possible around the REST API docs and only change things if there's a need for it!

File: community/kernel/src/main/java/org/neo4j/kernel/impl/annotations/AnnotationProcessor.java [hunk 2]
```diff
@@ -40,25 +36,11 @@
 import javax.tools.FileObject;
 import javax.tools.StandardLocation;
 
-import static org.neo4j.kernel.impl.util.Charsets.UTF_8;
 import static org.neo4j.io.fs.FileUtils.newFilePrintWriter;
+import static org.neo4j.kernel.impl.util.Charsets.UTF_8;
```
Comment (nawroth) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/annotations/AnnotationProcessor.java:40]: What's up with the changes around charset? 
See https://github.com/neo4j/neo4j/commit/b13d1b2980548fe4e5d9a633b3457b354d8d19dd#diff-d232b77c0bec2f58f44f5193f4bf3293
This is the only conflict when forward-merging.

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/annotations/AnnotationProcessor.java:40]: @nawroth there is no more `org.neo4j.kernel.impl.util.Charsets.UTF_8` just change those to `java.nio.charset.StandardCharsets.UTF_8`

**Pull request merged**